File: app/src/main/java/com/mdkashif/alarm/utils/db/AppDatabase.java
package com.mdkashif.alarm.utils.db;

import android.content.Context;

import com.mdkashif.alarm.alarm.miscellaneous.DaoAccess;
import com.mdkashif.alarm.alarm.miscellaneous.TimingsModel;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {TimingsModel.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase mINSTANCE;

    public abstract DaoAccess userDao();

    public static AppDatabase getAppDatabase(Context context) {
        if (mINSTANCE == null) {
            mINSTANCE =
                    Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "universal-alarm-db")
                            .allowMainThreadQueries()
                            .build();
        }
        return mINSTANCE;
    }

    public static void destroyInstance() {
        mINSTANCE = null;
    }
}

File: app/src/main/java/com/mdkashif/universalarm/persistence/AppDatabase.kt
package com.mdkashif.universalarm.persistence

import androidx.room.Database
import androidx.room.RoomDatabase
import com.mdkashif.universalarm.alarm.misc.model.DaysModel
import com.mdkashif.universalarm.alarm.misc.model.LocationsModel
import com.mdkashif.universalarm.alarm.misc.model.TimingsModel

@Database(entities = [TimingsModel::class, DaysModel::class, LocationsModel::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun accessDao(): RoomAccessDao
}
